### DISCUSSION TITLE
rtp needed for project-local nvim approach?

### DISCUSSION THREAD

**Discussion by stefanvanburen** [Olical/nfnl#41] [Q&A] [answered]
Hey! I'm looking to follow your approach in [suggested here](https://github.com/Olical/nfnl?tab=readme-ov-file#directory-local-neovim-configuration-in-fennel) for project local nvim configuration.

I've enabled `exrc`, then created a `.nfnl.fnl` file:

```fennel
;; .nfnl.fnl
{:source-file-patterns [:.nvim.fnl]}
```

Then I create a `.nvim.fnl` file:
```fennel
;; .nvim.fnl
(print "hello there!")
```

I've confirmed that it compiles the `.nvim.fnl` file to `.nvim.lua`, but then when I load another file in the project (e.g. `nvim README.md`), it doesn't seem to load the `.nvim.lua` file (I never see anything printed 😄). I suspect I'm missing something obvious about adding that file to the `rtp`, or something, but thought I'd ask — what's your approach to ensuring that the compiled lua is loaded?

**Comment by stefanvanburen** [comment]
Actually, realizing that setting `exrc` [should automatically load the `.nvim.lua` file](https://neovim.io/doc/user/options.html#'exrc') — hmm. Probably an issue on my end! Going to dig in more, but if anything obvious is wrong let me know :).

**Reply by stefanvanburen** [reply, marked as answer]
For posterity: I wasn't enabling `exrc` (doing a `(set vim.g.exrc true)` instead of `(set vim.o.exrc true)`) — embarassing. Thanks for the great project!

**Reply by Olical** [reply]
Glad you worked it out!
